### DISCUSSION TITLE
Halving sample time for the S+H LFO

### DISCUSSION THREAD

**Discussion by manysounds** [SynthstromAudible/DelugeFirmware#421] [Ideas]
TLDR: S+H feels like it's half the speed of the other LFO shapes.

Perceptually the S+H rate seems to be half of the other LFO types. In most-use cases  I find myself doubling the rate of S+H to "feel" like the other waveforms. Each cycle in the Triangle or Saw and most obviously Square sound like two points, top and bottom. When switching to S+H there is only one point per cycle and it *feels* like the speed has been halved. By halving the sample time for S+H this gives the same perception of there being two points per cycle like every other LFO shape (aside from noise, obv).
It's an idea. :)

**Comment by manysounds** [comment]
Adding that this seems to be the case for the Drunk Walk LFO as well.

**Comment by ok-reza** [comment]
it's just the nature of how S&H and Random Walk work and is the expected rate behavior.
if S&H LFO rate is set to 1 bar, it will sample and hold a new value every 1 bar, just like if a saw LFO rate is set to 1 bar it will ramp from - to + once every bar. (i am not using the most up-to-date firmware, if this isn't the actual behavior currently occurring on your end, then maybe it's a bug)
i can understand your specific use case, but it would cause confusion and disconnect from the logic of traditional implementation of LFO shapes (AFAIK!)
